bluepysnap/nodes.py

```diff
@@ -239,9 +200,103 @@ def _check_ids(self, node_ids):
             raise BluepySnapError("node ID not found: [%s]" % ",".join(map(str, missing)))
 
     def _check_property(self, prop):
+        """Check if a property exists inside the dataset."""
         if prop not in self.property_names:
             raise BluepySnapError("No such property: '%s'" % prop)
 
+    def _get_node_set(self, node_set_name):
+        """Returns the node set named 'node_set_name'."""
+        if node_set_name not in self._node_sets:
+            raise BluepySnapError("Undefined node set: '%s'" % node_set_name)
+        res = self._node_sets[node_set_name]
+        return None if res == {} else res
+
+    def _positional_mask(self, node_ids):
+        """Positional mask for the node IDs.
+
+        Examples:
+            if the data set contains 5 nodes:
+            _positional_mask([0,2]) --> [True, False, True, False, False]
+        """
+        if node_ids is None:
+            return np.full(len(self._data), fill_value=True)
+        mask = np.full(len(self._data), fill_value=False)
+        mask[node_ids] = True
+        return mask
+
+    def _node_population_mask(self, queries):
+        """Handle the population and node ID queries."""
+        populations = queries.pop(POPULATION_KEY, None)
+        if populations is not None and self.name not in set(utils.ensure_list(populations)):
+            node_ids = []
+        else:
+            node_ids = queries.pop(NODE_ID_KEY, None)
+        return queries, self._positional_mask(node_ids)
+
+    def _properties_mask(self, queries):
+        """Return mask of node IDs with rows matching `props` dict."""
+        # pylint: disable=assignment-from-no-return
+        unknown_props = set(queries) - set(set(self._data.columns) | {POPULATION_KEY, NODE_ID_KEY})
```

Comment:
Maybe this is clearer ? (maybe ?)
```suggestion
        unknown_props = set(queries) - set(self._data.columns) - {POPULATION_KEY, NODE_ID_KEY}
```

Reply:
yes makes sense. Done